src/azureExtensions.ts

```diff
@@ -141,7 +141,7 @@ export const azureExtensions: IAzExtMetadata[] = [
 ];
 
 export const legacyTypeMap: Partial<Record<AzExtResourceType, string>> = {
-    AiFoundry: 'microsoft.machinelearningservices/workspaces',
+    AiFoundry: 'microsoft.cognitiveServices/accounts',
```

Comment:
Why has this been replaced and not just added?

Reply:
1. This doesn't support adding two values with same key.
2. The `legacyTypeMap` is referenced in `createGroupConfigFromResource` and `createAzureExtensionsGroupConfig`, neither of them is used. So, this change doesn't seem to take any effect actually. 

If one day the `legacyTypeMap` take effect and it can only map to one type, I think it should be the new type.